command/pipe.go

```diff
@@ -0,0 +1,360 @@
+	if err != nil {
+		printError(fullCommand, c.Command.Name, err)
+		return nil, err
+	}
+
+	return &Pipe{
+		dst:          dst,
+		op:           c.Command.Name,
+		fullCommand:  fullCommand,
+		deleteSource: deleteSource,
+		// flags
+		noClobber:             c.Bool("no-clobber"),
+		storageClass:          storage.StorageClass(c.String("storage-class")),
+		concurrency:           c.Int("concurrency"),
+		partSize:              c.Int64("part-size") * megabytes,
+		encryptionMethod:      c.String("sse"),
+		encryptionKeyID:       c.String("sse-kms-key-id"),
+		acl:                   c.String("acl"),
+		forceGlacierTransfer:  c.Bool("force-glacier-transfer"),
+		ignoreGlacierWarnings: c.Bool("ignore-glacier-warnings"),
+		cacheControl:          c.String("cache-control"),
+		expires:               c.String("expires"),
+		contentType:           c.String("content-type"),
+		contentEncoding:       c.String("content-encoding"),
+		dstRegion:             c.String("destination-region"),
+
+		// s3 options
+		storageOpts: NewStorageOpts(c),
+	}, nil
+}
+
+// Run starts copying stdin output to destination.
+func (c Pipe) Run(ctx context.Context) error {
+	var task = c.prepareUploadTask(ctx, c.dst)
+	var err = task()
+
+	return err
+}
+
+func (c Pipe) prepareUploadTask(
+	ctx context.Context,
+	dsturl *url.URL,
+) func() error {
+	return func() error {
+		if dsturl.IsBucket() || dsturl.IsPrefix() {
+			return fmt.Errorf("target %q must be a file", dsturl)
+		}
+
+		err := c.doUpload(ctx, dsturl)
+		if err != nil {
+			return &errorpkg.Error{
+				Op:  c.op,
+				Src: nil,
+				Dst: dsturl,
+				Err: err,
+			}
+		}
+		return nil
+	}
+}
+
+func (c Pipe) doUpload(ctx context.Context, dsturl *url.URL) error {
+	srcClient := storage.NewLocalClient(c.storageOpts)
+
+	stdinReader, err := srcClient.ReadStdin()
+	if err != nil {
+		return err
+	}
+
+	err = c.shouldOverride(ctx, dsturl)
+	if err != nil {
+		if errorpkg.IsWarning(err) {
+			printDebug(c.op, err, nil, dsturl)
+			return nil
+		}
+		return err
+	}
+
+	// override destination region if set
+	if c.dstRegion != "" {
+		c.storageOpts.SetRegion(c.dstRegion)
+	}
+	dstClient, err := storage.NewRemoteClient(ctx, dsturl, c.storageOpts)
+	if err != nil {
+		return err
+	}
+
+	metadata := storage.NewMetadata().
+		SetStorageClass(string(c.storageClass)).
+		SetSSE(c.encryptionMethod).
+		SetSSEKeyID(c.encryptionKeyID).
+		SetACL(c.acl).
+		SetCacheControl(c.cacheControl).
+		SetExpires(c.expires)
+
+	if c.contentType != "" {
+		metadata.SetContentType(c.contentType)
+	} else {
+		metadata.SetContentType(guessStdinContentType(dsturl))
+	}
+
+	if c.contentEncoding != "" {
+		metadata.SetContentEncoding(c.contentEncoding)
+	}
+
+	err = dstClient.Put(ctx, stdinReader, dsturl, metadata, c.concurrency, c.partSize)
+	if err != nil {
+		return err
+	}
+
+	msg := log.InfoMessage{
+		Operation:   c.op,
+		Source:      nil,
+		Destination: dsturl,
+		Object: &storage.Object{
+			StorageClass: c.storageClass,
+		},
+	}
```

Comment:
```bas
; echo 'stuff' | ./s5cmd pipe s3://bucket/selman/stuff-test.txt
pipe <nil> s3://bucket/selman/stuff-test.txt
```

Looks like `log.InfoMessage.String()` does not handle this case.

Reply:
Done. https://github.com/peak/s5cmd/pull/587/commits/3c5c712588edada2b2813559beb9cb05c317868c